examples/msgsend/main.rs

```diff
@@ -37,18 +38,18 @@ fn server(mut pull_socket: zmq::Socket, mut push_socket: zmq::Socket, mut worker
     }
 }
 
-fn spawn_server(ctx: &mut zmq::Context, workers: u64) -> Sender<()> {
-    let mut pull_socket = ctx.socket(zmq::PULL).unwrap();
-    let mut push_socket = ctx.socket(zmq::PUSH).unwrap();
-
-    pull_socket.bind("inproc://server-pull").unwrap();
-    push_socket.bind("inproc://server-push").unwrap();
-
+fn spawn_server(ctx: Arc<zmq::Context>, workers: u64) -> Sender<()> {
     // Spawn the server.
     let (ready_tx, ready_rx) = channel();
     let (start_tx, start_rx) = channel();
 
     thread::spawn(move|| {
+        let mut pull_socket = ctx.socket(zmq::PULL).unwrap();
+        let mut push_socket = ctx.socket(zmq::PUSH).unwrap();
+
+        pull_socket.bind("inproc://server-pull").unwrap();
+        push_socket.bind("inproc://server-push").unwrap();
+
```

Comment:
By using Arc for the context and moving the socket creation into the thread, the compiler automatically pulls the Arc pointer into the thread context, ensuring it lives long enough.